### PR TITLE
Implement database schema

### PR DESCRIPTION
**Pull request by amatranga**
Updated schema is located in db/migrations. Updating the schema required
updates to many files including...

db/seeds/test/users_seed.js --- Add firstName field to test users
server/views/signup.ejs--Add field for name
server/middleware/passport.js--Update to expect a name
knexfile.js--Added a catch-all case (if for some reason you are using a Windows computer or your env is messed up)
package.json--Add test and run commands for Windows computers

### CODE REVIEW

File: server/middleware/passport.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ passport.deserializeUser((id, done) => {
 });
 
 passport.use('local-signup', new LocalStrategy({
+  firstnameField: 'fisrt',
```
Comment (Schleppguy) [inline server/middleware/passport.js]: Fix 'fisrt' to be 'first'

File: db/migrations/20170326215143_initial.js [hunk 2]
```diff
@@ -1,30 +1,74 @@
...
   return Promise.all([
-    knex.schema.createTableIfNotExists('profiles', function (table) {
+    knex.schema.createTableIfNotExists('profiles', function(table){
       table.increments('id').unsigned().primary();
-      table.string('first', 100).nullable();
+      table.string('first', 100).notNullable();
       table.string('last', 100).nullable();
       table.string('display', 100).nullable();
-      table.string('email', 100).nullable().unique();
+      table.string('email', 100).notNullable();
```
Comment (Schleppguy) [inline db/migrations/20170326215143_initial.js]: Email needs to be unique

File: db/seeds/test/users_seed.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ exports.seed = function (knex, Promise) {
       return models.Profile.forge({
         first: 'System',
         last: 'Admin',
+        password: 'admin123',
```
Comment (Schleppguy) [inline db/seeds/test/users_seed.js]: No password in Profile, correct?

File: server/middleware/passport.js (hunk 1 above)
Comment (Schleppguy) [reply server/middleware/passport.js]: this doesn't need to be here at all actually

File: server/middleware/passport.js [hunk 4]
```diff
@@ -70,6 +71,7 @@ passport.use('local-signup', new LocalStrategy({
   }));
 
 passport.use('local-login', new LocalStrategy({
+  firstnameField: 'first',
```
Comment (Schleppguy) [inline server/middleware/passport.js]: This doesn't need to be here. Can remove this line.

File: db/migrations/20170326215143_initial.js [hunk 5]
```diff
@@ -1,30 +1,74 @@
-
-exports.up = function (knex, Promise) {
+exports.up = function(knex, promise){
```
Comment (Schleppguy) [inline db/migrations/20170326215143_initial.js]: This should be a new file. Look into how to create a migration in knex.

**Review by Schleppguy — CHANGES_REQUESTED** [review]

File: server/middleware/passport.js (hunk 1 above)
Comment (amatranga) [reply server/middleware/passport.js]: Removed firstnameField

File: server/middleware/passport.js (hunk 4 above)
Comment (amatranga) [reply server/middleware/passport.js]: removed

File: db/migrations/20170326215143_initial.js (hunk 5 above)
Comment (amatranga) [reply db/migrations/20170326215143_initial.js]: Moved into a new file

File: db/migrations/20170326215143_initial.js (hunk 2 above)
Comment (amatranga) [reply db/migrations/20170326215143_initial.js]: updated with migration change

File: db/seeds/test/users_seed.js (hunk 3 above)
Comment (amatranga) [reply db/seeds/test/users_seed.js]: removed

**Review by Schleppguy — APPROVED** [review]

**Pull request merged**